packages/optimizely-cms-sdk/src/model/displayTemplates.ts

```diff
@@ -62,20 +62,26 @@ export type DisplayTemplate<T = DisplayTemplateVariant> = T & {
   __type: 'displayTemplate';
 };
 
+
 export function parseDisplaySettings(
-  displaySettings?: DisplaySettingsType[] | null
-): Record<string, string> | undefined {
-  if (!displaySettings) {
-    return undefined; // Return undefined if displaySettings is not provided
-  }
+  settings: DisplaySettingsType[] | null | undefined,
+  templateSettings: Record<string, { editor: string; choices: Record<string, any> }>
+): Record<string, string> {
+  if (!settings || settings.length === 0) return {};
 
-  const result: Record<string, string> = {}; // Initialize an empty object
+  const result: Record<string, string> = {};
 
-  // Iterate over the input array
-  for (const item of displaySettings) {
-    // Assign the value to the key in the result object
-    result[item.key] = item.value;
-  }
+  settings.forEach(s => {
```

Comment:
On the second thought we might not need to do this since parsing. Lets stick to the older way where we just did something like this

```
  for (const item of displaySettings) {
    // Assign the value to the key in the result object
    result[item.key] = item.value;
  }
```